### PR TITLE
Lesson 3

### PR DESCRIPTION
**Pull request by juanvi-martinez**
Documentation for lesson 3

### CODE REVIEW

File: 03/README.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+3. **Attribute(s)/property(ies):** keep state
+4. **Method(s)/function(s):** Do something
+
+##### Visibility modifiers in Kotlin
+Classes, objects, interfaces, constructors, functions, properties and their setters can have visibility modifiers.
+
+- `private` - visible (can be accessed) from inside the class only.
+- `protected` - visible to the class and its subclass.
+- `internal` - any client inside the module can access them.
+- `public` - visible everywhere.
```
Comment (vvalouch) [inline 03/README.md:40]: Could we have some task that is playing with accessibility modifiers?

File: 03/README.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    - Add also 2 new properties: `name` and `maxFuel`
+    - `name`, `maxFuel` and `maxSpeed` are constructor parameters
+
+### Interfaces
+An interface generally defines the set of methods/functions along with their arguments and applicable return types that an instance of a class that has that interface could respond to. How those set of operations are implemented, is left to that particular class. 
+
+
+#### Task
+
+1. Create an interface called `CarInterface` for our `Car` class
```
Comment (vvalouch) [inline 03/README.md]: Matter of taste but CarInterface could be just Car and current car could be named SportsCar

File: 03/README.md (hunk 1 above)
Comment (juanvi-martinez) [reply 03/README.md:40]: yes, i'm planning to use different accessibility modifiers in functions and properties while working on `Task 2` of `Working with classes` section 🙂

File: 03/README.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+### Abstract classes
+Abstract classes are classes that contain one or more abstract methods. An abstract method is a method that is declared, but contains no implementation. Abstract classes may not be instantiated (you cannot create objects of an abstract class), and require subclasses to provide implementations for the abstract methods.
+
+#### Task
+1. Create an abstract class called `CarAbstract` for our `Car` class
```
Comment (vvalouch) [inline 03/README.md]: Analogy in naming might be used: CarAbstract => Volkswagen Group MEB platform, Car => Skoda Karoq; CarInterface => expectations from any car

File: 03/README.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    ```
+7. Right click -> run `MainKt` and check that it works
+
+
+### Working with classes
+#### Task 1
+1. Convert the previous `Hello world` implementation to an Objected-oriented version
+
+#### Task 2
+1. Define the class `Car` according to the image above
```
Comment (vvalouch) [inline 03/README.md:76]: This task might be actually split to multiple if you would like to show how to use primary, secondary constructors, init... Each class could be different car model. E.g. FabiaCar for simplest one and SuperbCar for most complex class ;-)

**Review by vvalouch — APPROVED** [review]
Looks good to me. I left few optional enhancement comments.

File: 03/README.md (hunk 4 above)
Comment (juanvi-martinez) [reply 03/README.md:76]: i was considering that, but i think it would be too much for this lesson :-)

**Pull request merged**